File: 50 Coding Challenge Part II/Python/Challenge49.py
# Challenge 49: Shuffle an array of strings
import random


def shuffleArray(arr):
    n = len(arr) - 1
    for i in range(n):
        random_index = random.randint(0, n)
        temp = arr.pop(random_index)
        arr.append(temp)
    return arr


# Driver Method
def main():
    ar = ["Shuffle", "an", "array", "of", "strings"]
    print(shuffleArray(ar))


if __name__ == "__main__":
    main()

File: 50 Coding Challenge Part II/TypeScript/Challenge49.ts
// Challenge 49: Shuffle an array of strings

// using math.random

function shuffleArray(arr: string[]) {
    let i = arr.length;
    while (--i > 0) {
        let randIndex = Math.floor(Math.random() * (i + 1));
        [arr[randIndex], arr[i]] = [arr[i], arr[randIndex]];
    }
    return arr;
}


// Driver Method
let ar = ["Shuffle", "an", "array", "of", "strings"]
console.log(shuffleArray(ar))
